File: PE10/PE10.py
# Slower than C++ and Java, takes like 30 s to run. Could optimize with sieve of erastothenes, but this is easier to program.

def isprime(n):
    if (n < 2):
        return False
    if (n < 4):
        return True
    for f in range(2, int(n**0.5)+1):
        if (n % f == 0):
            return False
    return True

mysum = 2
upto = 2000000
for i in range(3, upto, 2):
    if (isprime(i)):
        mysum += i
print(mysum)

File: PE10/PE10.java

public class PE10 {
	public static boolean isprime(int n) {
		if (n > 1 && n < 4) return true;
		for (int f = 2; f <= (int)Math.sqrt(n); f++) {
			if (n % f == 0) return false;
		}
		return true;
	}
	
	public static void main(String args[]) {
		long mysum = 2;
		int upto = 2000000;
		for (int i = 3; i < upto; i += 2) {
			if (isprime(i)) {
				mysum += i;
			}
		}
		System.out.println(mysum);
	}
}
